### PR TITLE
Search for in-progress new registrations

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-154

Logged-in users should be able to choose between searching existing registrations or ones which are still in progress (transient registrations). The results should be displayed in a way that is consistent with active registrations, with missing data dealt with gracefully.

### CODE REVIEW

**Comment by irisfaraway** [comment]
I disagree with CodeClimate on this one – think the duplication is necessary in this case.

File: app/models/waste_exemptions_engine/address.rb [hunk 1]
```diff
@@ -4,10 +4,6 @@
 
 module WasteExemptionsEngine
   class Address
-    scope :search_for_postcode, lambda { |term|
-      where("UPPER(postcode) LIKE ?", "%#{term&.upcase}%")
-    }
-
-    scope :site, -> { where(address_type: 3) }
+    include WasteExemptionsEngine::CanBeSearchedLikeAddress
```
Comment (Cruikshanks) [inline app/models/waste_exemptions_engine/address.rb]: I don't think it needs the Module name

```suggestion
    include CanBeSearchedLikeAddress
```

I could be wrong 😁

File: app/models/waste_exemptions_engine/transient_address.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require WasteExemptionsEngine::Engine.root.join("app", "models", "waste_exemptions_engine", "transient_address")
+
+module WasteExemptionsEngine
+  class TransientAddress
+    include WasteExemptionsEngine::CanBeSearchedLikeAddress
```
Comment (Cruikshanks) [inline app/models/waste_exemptions_engine/transient_address.rb]: I don't think it needs the Module name

```suggestion
    include CanBeSearchedLikeAddress
```

Again I could be wrong 😀

File: app/models/waste_exemptions_engine/address.rb (hunk 1 above)
Comment (irisfaraway) [reply app/models/waste_exemptions_engine/address.rb]: You are correct! I got it right for the others so not sure why I did it differently here 🤔

**Comment by Cruikshanks** [comment]
I'm also happy if you want to mark the codeclimate issues as `Wontfix` in this case.

**Pull request merged**